drgn/internal/cli.py

```diff
@@ -176,6 +176,8 @@ def main() -> None:
     init_globals: Dict[str, Any] = {"prog": prog}
     if args.script:
         sys.argv = args.script
+        if os.path.isfile(args.script[0]):
```

Comment:
This is an obscure corner case, but if we want exactly the same behavior as the `python` command, then we don't want to do this for zip files. For directories and zip files, the path itself is inserted, not the parent directory (and `runpy` already does that: https://github.com/python/cpython/blob/4e796f56465f7264ddba63c8396d2649e098b617/Lib/runpy.py#L280).

So cargo culting the `runpy` code, maybe we want this:

```suggestion
        import pkgutil
        if pkgutil.get_importer(args.script[0]) is None:
```

I'm not sure whether it's worth doing that.